[Doc] Modify SECURITY.md nested dependency fix and add backport process (#3497)
opensearch-project/OpenSearch-Dashboards: CHANGELOG.md

- Updates NOTICE file, adds validation to GitHub CI ([#3051](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3051))
- [Doc] Add current plugin persistence implementation readme ([#3081](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3081))
- Correct copyright date range of NOTICE file and notice generator ([#3308](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3308))

> - [Doc] Modify SECURITY.md nested dependency fix and add backport process ([#3497](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3497))

Reviewer: ```suggestion
- [Doc] Update SECURITY.md with instructions for nested dependencies and backporting ([#3497](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3497))
```

Author: Will change
